example/basic/basic_hcube.cxx

```diff
@@ -150,17 +205,10 @@ main (int argc, char **argv)
   opt = sc_options_new (argv[0]);
   sc_options_add_switch (opt, 'h', "help", &helpme,
                          "Display a short help message.");
-  sc_options_add_int (opt, 'l', "level", &level, 0,
-                      "The refinement level of the mesh.");
-  sc_options_add_switch (opt, 'p', "partition", &do_partition,
-                         "Enable coarse mesh partitioning.");
+  sc_options_add_int (opt, 'd', "dimension", &dim, 1,
+                      "The dimension of the mesh.");
   sc_options_add_switch (opt, 'b', "balance", &do_balance,
                          "Additionally balance the forest.");
```

Comment:
If the user does not have a choice for the maximum refinement level (which is okay, since it remove complexity), then we do not need the maximum refinement level as user data of the forest and should remove it.
Instead we can just set the maximum level variable in the refine callback to 5.
Thus, we remove again a piece of complexity.